operators/src/processing/expression.rs

```diff
@@ -70,21 +69,52 @@ impl RasterOperator for Expression {
                 found: self.vector_sources.len()
             }
         );
+        ensure!(
+            !self.raster_sources.is_empty(),
+            crate::error::InvalidNumberOfVectorInputs {
+                expected: 1..9,
+                found: self.raster_sources.len()
+            }
+        );
 
-        InitializedOperatorImpl::create(
-            self.params,
-            context,
-            |_, _, _, _| Ok(()),
-            |params, _, _, _, _| {
-                Ok(RasterResultDescriptor {
-                    data_type: params.output_type,
-                    spatial_reference: SpatialReferenceOption::Unreferenced, // TODO
-                })
-            },
-            self.raster_sources,
-            vec![],
+        let raster_sources = self
+            .raster_sources
+            .into_iter()
+            .map(|source| source.initialize(context))
+            .collect::<Result<Vec<_>>>()?;
+
+        let spatial_reference = raster_sources[0].result_descriptor().spatial_reference;
+
+        for other_spatial_refenence in raster_sources
+            .iter()
+            .skip(1)
+            .map(|source| source.result_descriptor().spatial_reference)
+        {
+            ensure!(
+                spatial_reference == other_spatial_refenence,
+                crate::error::InvalidSpatialReference {
+                    expected: spatial_reference,
+                    found: other_spatial_refenence,
+                }
+            );
+        }
+
+        let result_descriptor = RasterResultDescriptor {
+            data_type: self.params.output_type,
+            spatial_reference,
+            measurement: raster_sources[0].result_descriptor().measurement.clone(), // TODO: use parameter to define it
```

Comment:
please do like output_type

Reply:
This breaks the expression (UI, etc.), so the task for it is in the expression story.

Reply:
d2c23f1